tracing-journald/src/memfd.rs

```diff
@@ -0,0 +1,42 @@
+//! memfd helpers.
+
+use libc::*;
+use std::ffi::CString;
+use std::fs::File;
+use std::io::Result;
+use std::io::{Error, ErrorKind};
+use std::os::raw::c_uint;
+use std::os::unix::prelude::{AsRawFd, FromRawFd, RawFd};
+
+fn create<S: AsRef<str>>(name: S, flags: c_uint) -> Result<File> {
+    let cname = CString::new(name.as_ref()).map_err(|_| {
+        Error::new(
+            ErrorKind::InvalidData,
+            format!("Name {:?} contains internal NUL byte", name.as_ref()),
+        )
+    })?;
+    unsafe {
+        let fd = memfd_create(cname.as_ptr(), flags);
```

Comment:
We only ever pass in a single `name` here, and according to the memfd_create manpage the name has no semantic meaning. Let's use a hardcoded constant (e.g. `b"tracing-journald\0"`) rather than creating a new `CString` on every call.

Reply:
Done.